src/ripple/ripple.js

```diff
@@ -118,7 +119,12 @@ MaterialRipple.prototype.upHandler_ = function(event) {
 
   // Don't fire for the artificial "mouseup" generated by a double-click.
   if (event && event.detail !== 2) {
-    this.rippleElement_.classList.remove(this.CssClasses_.IS_VISIBLE);
+    // Allow a repaint to occur before removing this class, so the animation
+    // shows for tap events, which seem to trigger a mouseup too soon after
+    // mousedown.
+    window.setTimeout(function() {
```

Comment:
Ok so in the react-animation docs they mention a "next-tick" approach..
https://github.com/facebook/react/blame/master/docs/docs/10.1-animation.md#L59

After a bit of digging into the implementation details here is the line in question

https://github.com/facebook/react/blob/34c8b2366ca4137b0a59f679fa12ae676da76119/src/addons/transitions/ReactCSSTransitionGroupChild.js#L95-L101

So it looks like `TICK` is a magical 17.... :rabbit2: for their setTimeout

So to be clear, the current approach is inline with the next-tick approach :gem: , but there is still a question of the correct timeout and being consistent internally. It has been my understanding that a timeout of 0 will accomplish the goal here of dequeuing and enqueuing the task to the bottom of the stack... but perhaps there is something here..... :cactus: